### DISCUSSION TITLE
About parties - a special case

### DISCUSSION THREAD

**Discussion by thommignot** [microsoft/electionguard#296] [Q&A]
Hello everybody, I'm facing a somewhat painful issue and can't figure out if there's a solution I could implement to it.

In France, there's something that could be translated as "Works council" I guess, and the workers vote for their delegates.
It has the following rules:
- there are a number of place, which is also the number of votes allowed
- candidates are part of a list (let's say each party have a list of candidate so: each candidate is from a party)
- a list is (usually) a list of `votes_allowed` candidates (let's say always for now) 
- one can give his votes to a whole list (all the candidates) or a subset of it (undervoting is allowed)

and here is the hard part:
- a ballot should be valid only if all the selected candidates are from the same party
- the tally must include how many individual ballots a party got. (not depending on how many votes the candidates have)

I know that electionguard alone cannot for handle this for the moment, but I'd be really grateful if anyone has a clue to solve this!

I'd want to implement it in such a way that allows proving the validity of a ballot and counting votes for individual parties.

We can think of such a construction that all possible choices are a "candidate":
if A is a party and the associated candidates are a, b, c so we have the following candidates:
`[A_abc, A_ab, A_ac, A_bc, A_a, A_b, A_c]` with `A_bc` representing "1 vote for list A, 1 for candidate `b` and 1 for candidate `c`"
But as you guess, computation is really heavy if we have more that ~6 candidates per list
But that the way I'm making it now. 

For now, the only solution I have in mind to scale it up is using some kind of re-encryption mixnet (adding encrypted zeros to the selections data I suppose) to have better anonymity when the ballots are sent but then decrypting individual ballots to check their validity before adding them to the tally or counting plaintext ballots.
Or I can avoid encrypting the ballots client side. I could check the validity of the form POST data before encrypting it on the serverside
 
Does anyone have an idea?

Thanks a lot to the electionguard team.

**Comment by felixdoerre** [comment]
If you're not bound by the electionguard format of ballots you could create the following ballot structure:
Have a ciphertext per "party", indicating which party the voter voted for. Add proofs that each ciphertext is 0 or 1 and their sum is 1.
For each individual list have the "regular" ZK proofs that the individual votes are 0 or 1.
Additionally, for each list: ZK-prove that the the sum of all votes for that list is 0 or that the "party"-ciphertext is 1.

I hope that gives you a rough idea.
